src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -145,22 +157,93 @@ public boolean matchCacheableRequest(
+                !cacheEntry.isValidationRequired();
 
     }
 
 
     public void purge(
         int requestHash)
     {
-        DefaultCacheEntry cacheEntry = this.cachedEntries.remove(requestHash);
-        entryCount.accept(-1);
+        DefaultCacheEntry cacheEntry = cachedEntries.remove(requestHash);
         if (cacheEntry != null)
         {
+            entryCount.accept(-1);
             cacheEntry.purge();
         }
+        links.remove(requestHash);
+    }
+
+    public void invalidateCacheEntryIfNecessary(
+        int requestHash,
+        String requestUrl,
+        ArrayFW<HttpHeaderFW> headers)
+    {
+        DefaultCacheEntry cacheEntry = cachedEntries.remove(requestHash);
+        if (cacheEntry != null)
+        {
+            cacheEntry.setValidationRequired(true);
+        }
+
+        headers.forEach(header ->
+        {
+            invalidateLinkCacheEntry(requestUrl, header);
+        });
     }
 
+    private void invalidateLinkCacheEntry(
+        String requestUrl,
+        HttpHeaderFW header)
+    {
+        if (LINK.equals(header.name().asString()))
+        {
+            Matcher matcher = LINK_URL_PATTERN.matcher(header.value().asString());
+            if (matcher.find())
+            {
+                String linkFullURL = matcher.group(1);
+                if (linkFullURL != null)
+                {
+                    try
+                    {
+                        final URI requestURI = new URI(requestUrl);
+                        if (URL_PATTERN.asPredicate().test(linkFullURL))
+                        {
+                            final URI linkRequestURI = new URI(linkFullURL);
+                            if (!requestURI.getAuthority().equals(linkRequestURI.getAuthority()))
+                            {
+                                return;
+                            }
+                        }
+                        else
+                        {
+                            linkFullURL = String.format("%s://%s%s",
+                                                        requestURI.getScheme(),
+                                                        requestURI.getAuthority(),
+                                                        linkFullURL);
+                        }
+
+                        final String finalLinkFullURL = linkFullURL;
+                        links.entrySet()
+                             .stream()
+                             .filter(map -> map.getValue().startsWith(finalLinkFullURL))
+                             .forEach(entry ->
+                             {
+                                 DefaultCacheEntry cacheEntry = get(entry.getKey());
+                                 if (cacheEntry != null)
+                                 {
+                                     cacheEntry.setValidationRequired(true);
+                                 }
+                             });
```

Comment:
This is a linear search of all URLs in the cache to find matches by link prefix.
Recommend changing the data structure to a `TreeMap<URI, CacheEntry>` so that it is sorted by `URI` with a comparator that preserves the sort order you need.
Then you can use `NavigableMap.tailMap(linkTargetURI)` to iterate matching cache entries efficiently.